artemis-server/src/main/java/org/apache/activemq/artemis/core/server/impl/ServerConsumerImpl.java

```diff
@@ -403,6 +404,13 @@ public HandleStatus handle(final MessageReference ref) throws Exception {
          return HandleStatus.BUSY;
       }
 
+      if (!server.callBrokerMessagePluginsCanAccept(this, ref)) {
```

Comment:
This is hot path. Needs to be a simple bool check if it should go into this code area e.g. hasBrokerMessagePlugins which shoild be a boolen flag behind it not a list